accounts/abi/abi.go

```diff
@@ -137,11 +137,11 @@ func (abi *ABI) UnmarshalJSON(data []byte) error {
 
 // MethodById looks up a method by the 4-byte id
 // returns nil if none found
-func (abi *ABI) MethodById(sigdata []byte) *Method {
+func (abi *ABI) MethodById(sigdata []byte) (*Method, error) {
 	for _, method := range abi.Methods {
 		if bytes.Equal(method.Id(), sigdata[:4]) {
-			return &method
+			return &method, nil
 		}
 	}
-	return nil
+	return nil, fmt.Errorf("ABI spec does not contain method signature in data: 0x%x", sigdata[:4])
```

Comment:
Generally Go uses shorter error messages since they tend to get appended to one another. If possible, try to make it a bit more concise. E.g. `fmt.Errorf("no method with id %#x", sigdata[:4])`

Note `0x%x` is the same at `%#x` for formatting.